R/ceb.R

```diff
@@ -0,0 +1,126 @@
+#' @title Calculate mean children ever born & living
+#'
+#' @description Use default arguments to calculate mean children ever born
+#' related statistics from a DHS individual recode file.
+#'
+#' @param data \[`data.frame()`\]\cr
+#'   Individual recode dataset.
+#' @param by \[`formula()`\]\cr
+#'   A formula specifying factor variables by which to stratify analysis.
+#'   Default is 'NULL' to specify no stratification. Passed to [`survey::svyby()`].
+#' @param agegr \[`numeric()`\]\cr
+#'   The start of each age group (in years) to group each individual's age by
+#'   and to stratify analysis by. Can be 'NULL' to specify no age-groups should
+#'   be created and stratified by. Default is 15, 20, 25, 30, 35, 40, 45, 50.
+#' @param clusters \[`formula()`\]\cr
+#'   A formula specifying cluster ids to be passed as `ids` to survey::svydesign.
+#'   Must correspond to variables in `data`. Default is '~v021'.
+#' @param strata \[`formula()`\]\cr
+#'   A formula specifying strata to be passed as `strata` to survey::svydesign.
+#'   Must correspond to variables in `data`. Default is '~v024+v025'.
+#' @param weight \[`character(1)`\]\cr
+#'   Variable name for sample weight. Default is 'v005'.
+#' @param ceb \[`character()`\]\cr
+#'   Variable names for children ever born related columns to calculate the mean
+#'   & standard error of. Value corresponds to name in `data`, name corresponds
+#'   to new variable name in output. Default is 'c(ceb = "v201", lc = "v218")'.
+#' @param dob \[`character(1)`\]\cr
+#'   Variable name for date of birth of each individual. Default is 'v008'.
+#'   Required when `agegr` is not 'NULL'.
+#' @param intv \[`character(1)`\]\cr
+#'   Variable name for interview date for each individual. Default is 'v008'.
+#'   Required when `agegr` is not 'NULL'.
+#' @param varmethod \[`character(1)`\]\cr
+#'   Method for variance calculation. Default is 'lin' for Taylor linearisation.
+#'   Other options include 'JKn' for stratified designs & 'JK1' for stratified
+#'   designs. See `survey::as.svrepdesign` for more information.
+#' @param scale \[`numeric(1)`\]\cr
+#'   Scale for dates inputs (`dob`, `intv`) to calendar years. 12 for CMC inputs.
+#'
+#' @seealso [survey::svydesign()], [survey::as.svrepdesign], [survey::svyby()],
+#' & [survey::svymean()] which this function uses internally.
+#'
+#' @return \[`data.frame()`\] consisting of estimates and standard errors as
+#' returned by `survey::svyby`. The full covariance matrix of the estimates can
+#' be retreived by `vcov(val)`.
+#'
+#' @examples
+#' data(zzir)
+#' calc_ceb(zzir)
+#'
+#' @export
+calc_ceb <- function(data,
+                     by = NULL,
+                     agegr = 3:10*5,
+                     clusters = ~v021,
+                     strata = ~v024+v025,
+                     weight = "v005",
+                     ceb = c(ceb = "v201", lc = "v218"),
+                     dob = "v011",
```

Comment:
Given these surveys have their own weird format for dates, do we need a special way to parse them?

Reply:
Yup, if you haven't seen a good explanation of century month code cmc format yet I think this description is good https://www.dhsprogram.com/Data/Guide-to-DHS-Statistics/Organization_of_DHS_Data.htm?rhtocid=_4_3_5#Construction_of_Variables.

The dob, agegr, intv variables are always numeric so don't need anything special to parse the value. The `scale` and `origin` arguments to `demogsurv` functions help with the logic internally within the functions.

Does this section make sense?
```
    data$agegr <- cut(
      x = data[[intv]] - data[[dob]],
      breaks = agegr*scale, labels = .epis_labels(agegr),
      include.lowest=TRUE, right=FALSE
    )
```
There is similar code here too for the other functions that group into age groups, cohorts or periods. https://github.com/ihmeuw-demographics/demogsurv/blob/master/R/demog_pyears.R

Reply:
It makes more sense now. thanks